### ISSUE TITLE
Convert `approx_median` to UDAF

### ISSUE THREAD

**Issue by jayzhan211** [apache/datafusion#10838]
### Is your feature request related to a problem or challenge?

Similar to https://github.com/apache/datafusion/pull/10713 https://github.com/apache/datafusion/pull/10836 https://github.com/apache/datafusion/pull/10834 and others in https://github.com/apache/datafusion/issues/8708

**Comment by goldmedal** [comment]
take

**Comment by goldmedal** [comment]
Hi @jayzhan211,

After some research, I found that `approx_median` heavily relies on `ApproxPercentileCont` for accumulation. 

https://github.com/goldmedal/datafusion/blob/1600a307f272f9c716131dcca7a55ba1761774a6/datafusion/physical-expr/src/aggregate/approx_median.rs#L46

I think we should convert `ApproxPercentileCont` to a UDAF first. We can move `ApproxPercentileCont` and `TDigest` to `function-aggregate`, so that `ApproxMedian` can use them.

WDYT?

**Comment by jayzhan211** [comment]
> approx_median

I agree, we need to convert `ApproxPercentileCont ` first. For `TDigest`,  we should move it to `datafusion/physical-expr-common/src/aggregate/mod.rs` where it can used not only for `functions-aggregate` but also other users that import `physical-expr-common`

**Comment by goldmedal** [comment]
> I agree, we need to convert `ApproxPercentileCont ` first. For `TDigest`, we should move it to `datafusion/physical-expr-common/src/aggregate/mod.rs` where it can used not only for `functions-aggregate` but also other users that import `physical-expr-common`

Thanks. I will have PR for it first.

**Comment by goldmedal** [comment]
Hi @jayzhan211
I tried to convert `ApproxPercentileCont`, but I encountered some challenges. `ApproxPercentileCont` creates the accumulator based on the `percentile` parameter, but I couldn't find a way to access this parameter when defining the UDAF.
https://github.com/goldmedal/datafusion/blob/24a08465e12bc07275cafe5310a7ac44898e39de/datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs#L104

The `AccumulatorArgs` doesn't carry the original argument expression, making it difficult to create the accumulator for `ApproxPercentileCont`.
https://github.com/goldmedal/datafusion/blob/24a08465e12bc07275cafe5310a7ac44898e39de/datafusion/expr/src/function.rs#L45

I think we need another PR to address this issue. For now, I just moved `TDigest` and `ApproxPercentileAccumulator` in #10840 so that I can convert `approx_median` to a UDAF.

**Comment by jayzhan211** [comment]
> Hi @jayzhan211 I tried to convert `ApproxPercentileCont`, but I encountered some challenges. `ApproxPercentileCont` creates the accumulator based on the `percentile` parameter, but I couldn't find a way to access this parameter when defining the UDAF. https://github.com/goldmedal/datafusion/blob/24a08465e12bc07275cafe5310a7ac44898e39de/datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs#L104
> 
> The `AccumulatorArgs` doesn't carry the original argument expression, making it difficult to create the accumulator for `ApproxPercentileCont`. https://github.com/goldmedal/datafusion/blob/24a08465e12bc07275cafe5310a7ac44898e39de/datafusion/expr/src/function.rs#L45
> 
> I think we need another PR to address this issue. For now, I just moved `TDigest` and `ApproxPercentileAccumulator` in #10840 so that I can convert `approx_median` to a UDAF.

We can add `self.args` to `AccumulatorArgs` and compute `percentile` and `tdigest max size` in `accumulator`.

**Comment by goldmedal** [comment]
> We can add `self.args` to `AccumulatorArgs` and compute `percentile` and `tdigest max size` in `accumulator`.
>

Thanks for the suggestion. I can make a PR after #10840.
